frappe/modules/utils.py

```diff
@@ -105,14 +103,31 @@ def sync(key, custom_doctype, doctype_fieldname):
 
 		# sync single doctype exculding the child doctype
 		def sync_single_doctype(doc_type):
-			frappe.db.sql('delete from `tab{0}` where `{1}` =%s'.format(
-				custom_doctype, doctype_fieldname), doc_type)
-			for d in data[key]:
-				if d.get(doctype_fieldname) == doc_type:
-					d['doctype'] = custom_doctype
-					doc = frappe.get_doc(d)
+			def _insert(data):
+				if data.get(doctype_fieldname) == doc_type:
+					data['doctype'] = custom_doctype
+					doc = frappe.get_doc(data)
 					doc.db_insert()
 
+			if custom_doctype != 'Custom Field':
+				frappe.db.sql('delete from `tab{0}` where `{1}` =%s'.format(
+					custom_doctype, doctype_fieldname), doc_type)
+
+				for d in data[key]:
+					_insert(data, key)
```

Comment:
check. too many arguments

Reply:
Fixed!

Reply:
@rmehta @saurabh6790 

it should be 
`_insert(d)`